homeassistant/components/sensor/gtfs.py

```diff
@@ -228,71 +301,201 @@ def unit_of_measurement(self):
     @property
     def device_state_attributes(self):
         """Return the state attributes."""
-        return self._attributes
+        return {k: self._attributes[k] for k in sorted(self._attributes)}
 
     @property
     def icon(self):
         """Icon to use in the frontend, if any."""
+        self._icon = ICONS.get(self._route_type, ICON)
         return self._icon
 
     def update(self):
         """Get the latest data from GTFS and update the states."""
         with self.lock:
+            # Fetch valid stop information once
+            if not self._origin:
+                stops = self._pygtfs.stops_by_id(self.origin)
+                if not stops:
+                    raise ValueError(
```

Comment:
What's the use of raising a ValueError if no one catches it? You should catch it and update the state accordingly. I suggest that if an error is raised, you return `False` from the `available` property.

Reply:
The goal was to error the component at start-up and prevent further attempts at fetching a non-existant database entry. However I can indeed exit this method gracefully; this way should the database record gets fixed between updates then the method will eventually provide a useful value.

Reply:
Fixed: now returning instead of throwing an exception.